### PR TITLE
Add possibility to pass config options to local node and down to init()

### PR DESCRIPTION
**Pull request by haadcode**
This PR will change the local() api so that options object can be passed to it.

### CODE REVIEW

**Comment by haadcode** [comment]
Fixed the tests. Ready to merge @VictorBjelkholm or @diasdavid?

**Comment by dignifiedquire** [comment]
@haadcode please rebase onto master

**Comment by haadcode** [comment]
I'm just gonna make a new, clean PR. Too much stuff in between has happened making it super hard to follow what actually changed here.

New PR here: https://github.com/ipfs/js-ipfsd-ctl/pull/123

**Comment by haadcode** [comment]
Closing this PR.
